src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -149,15 +147,52 @@ private void consume() throws Exception {
         continue;
       }
 
-      if (record == null)
-        continue;
+      if (record == null) continue;
 
       GenericRecord avroRecord = Utils.genericRecordFromJson(record.value());
       if (avroRecord == null) {
         _sensors._consumeError.record();
         continue;
       }
       int partition = record.partition();
+
+      /* Commit availability and commit latency service */
+      try {
+        TopicPartition topicPartition = new TopicPartition(_topic, partition);
+        final AtomicBoolean callbackFired = new AtomicBoolean(false);
+        final AtomicReference<Exception> offsetCommitIssues = new AtomicReference<>(null);
+        OffsetAndMetadata offsetAndMetadata;
+        /* Call commitAsync, wait for a NON-NULL return value (see https://issues.apache.org/jira/browse/KAFKA-6183) */
+        OffsetCommitCallback commitCallback = new OffsetCommitCallback() {
+          @Override
+          public void onComplete(Map<TopicPartition, OffsetAndMetadata> topicPartitionOffsetAndMetadataMap, Exception exception) {
+            if (exception != null) {
+              offsetCommitIssues.set(exception);
+            }
+            callbackFired.set(true);
+          }
+        };
+
+        if (_offsetsToCommit != null) {
+          _consumer.commitAsync(_offsetsToCommit, commitCallback);
+        } else {
+          _consumer.commitAsync(commitCallback);
+        }
+        _commitAvailabilityMetrics._offsetsCommitted.record();
+
+        org.testng.Assert.assertNull(offsetCommitIssues.get(), "Offset commit failed.");
```

Comment:
testng should not be used within actual production code.

If having a method that checks for a condition is more clear then perhaps the google guava library is the way to go.  https://github.com/google/guava/wiki/PreconditionsExplained  It's probably ok to add a dependency on that library KMF does not already have this dependency.

Reply:
Amended to not use testng library within the prod codebase.